sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -453,6 +494,47 @@ case class BroadcastHashJoinExec(
     val (keyEv, anyNull) = genStreamSideJoinKey(ctx, input)
     val (matched, checkCondition, _) = getJoinCondition(ctx, input)
     val numOutput = metricTerm(ctx, "numOutputRows")
+    val isLongHashedRelation = broadcastRelation.value.isInstanceOf[LongHashedRelation]
```

Comment:
Can you move this check to in the branch it is used ?

Reply:
done